test/unittest/dds/publisher/PublisherTests.cpp

```diff
@@ -196,14 +196,213 @@ TEST(PublisherTests, ChangeDefaultDataWriterQos)
+    qos.resource_limits().max_instances = 100;
+    qos.resource_limits().max_samples_per_instance = 500;
+    qos.resource_limits().allocated_samples = 50;
+    qos.resource_limits().extra_samples = 2;
+    // .transport_priority
+    qos.transport_priority().value = 10;
+    // .lifespan
+    qos.lifespan().duration.seconds = 10;
+    qos.lifespan().duration.nanosec = 33;
+    // .user_data
+    qos.user_data().push_back(0);
+    qos.user_data().push_back(1);
+    qos.user_data().push_back(2);
+    qos.user_data().push_back(3);
+    // .ownership
+    qos.ownership().kind = eprosima::fastdds::dds::EXCLUSIVE_OWNERSHIP_QOS;
+    // .ownership_strength
+    qos.ownership_strength().value = 30;
+    // .writer_data_lifecycle
+    qos.writer_data_lifecycle().autodispose_unregistered_instances = false;
+    // .publish_mode
+    qos.publish_mode().kind = eprosima::fastdds::dds::ASYNCHRONOUS_PUBLISH_MODE;
+    qos.publish_mode().flow_controller_name = "Prueba";
+
+    // .properties
+    eprosima::fastrtps::rtps::Property property;
+    property.name("Property1");
+    property.value("Value1");
+    qos.properties().properties().push_back(property);
+    property.name("Property2");
+    property.value("Value2");
+    qos.properties().properties().push_back(property);
+    // .reliable_writer_qos
+    qos.reliable_writer_qos().times.initialHeartbeatDelay.seconds = 2;
+    qos.reliable_writer_qos().times.initialHeartbeatDelay.nanosec = 15;
+    qos.reliable_writer_qos().times.heartbeatPeriod.seconds = 3;
+    qos.reliable_writer_qos().times.heartbeatPeriod.nanosec = 16;
+    qos.reliable_writer_qos().times.nackResponseDelay.seconds = 4;
+    qos.reliable_writer_qos().times.nackResponseDelay.nanosec = 17;
+    qos.reliable_writer_qos().times.nackSupressionDuration.seconds = 5;
+    qos.reliable_writer_qos().times.nackSupressionDuration.nanosec = 18;
+    qos.reliable_writer_qos().disable_positive_acks.enabled = true;
+    qos.reliable_writer_qos().disable_positive_acks.duration.seconds = 13;
+    qos.reliable_writer_qos().disable_positive_acks.duration.nanosec = 320;
+    qos.reliable_writer_qos().disable_heartbeat_piggyback = true;
+    // .endpoint
+    qos.endpoint().user_defined_id = 1;
+    qos.endpoint().entity_id = 2;
+    qos.endpoint().history_memory_policy = eprosima::fastrtps::rtps::PREALLOCATED_WITH_REALLOC_MEMORY_MODE;
+    // . writer_resource_limits
+    qos.writer_resource_limits().matched_subscriber_allocation.initial = 30;
+    qos.writer_resource_limits().matched_subscriber_allocation.maximum = 300;
+    qos.writer_resource_limits().matched_subscriber_allocation.increment = 400;
+    // . data_sharing
+    qos.data_sharing().on("/");
 
     ASSERT_TRUE(publisher->set_default_datawriter_qos(qos) == ReturnCode_t::RETCODE_OK);
     DataWriterQos wqos;
     publisher->get_default_datawriter_qos(wqos);
 
+    // .durability
+    ASSERT_EQ(eprosima::fastdds::dds::TRANSIENT_DURABILITY_QOS, wqos.durability().kind);
```

Comment:
I would change all the `ASSERT_XXX` into `EXPECT_XXX`

Reply:
Done